engine/dlib/src/dlib/dns.cpp

```diff
@@ -405,6 +405,7 @@ namespace dmDNS
             if ((dmTime::GetTime() - request_started) > request_timeout)
             {
                 req.m_Status = ARES_ETIMEOUT;
+                ares_cancel(dns_channel->m_Handle);
```

Comment:
When hitting the timeout, just after the host lookup has started, we were missing a ares_cancel.